sensor/kubernetes/certrefresh/utils_test.go

```diff
@@ -0,0 +1,138 @@
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+					Name:       ownerName,
+					UID:        "deployment-uid",
+				},
+			},
+		},
+	}
+}
+
+func TestFetchSensorDeploymentOwnerRef_Success(t *testing.T) {
+	k8sClient := fake.NewSimpleClientset(
+		createTestPod(sensorPodName, sensorNamespace, replicaSetName),
+		createTestReplicaSet(replicaSetName, sensorNamespace, deploymentName),
+	)
+
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), sensorPodName, sensorNamespace, k8sClient, backoff)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, ownerRef)
+	assert.Equal(t, "apps/v1", ownerRef.APIVersion)
+	assert.Equal(t, "Deployment", ownerRef.Kind)
+	assert.Equal(t, deploymentName, ownerRef.Name)
+	assert.Equal(t, types.UID("deployment-uid"), ownerRef.UID)
+}
+
+func TestFetchSensorDeploymentOwnerRef_EmptyPodName(t *testing.T) {
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), "", sensorNamespace,
+		fake.NewSimpleClientset(), backoff)
+
+	assert.Error(t, err)
+	assert.Nil(t, ownerRef)
+}
+
+func TestFetchSensorDeploymentOwnerRef_PodNotFound(t *testing.T) {
+
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), sensorPodName, sensorNamespace,
+		fake.NewSimpleClientset(), backoff)
+
+	assert.Error(t, err)
+	assert.Nil(t, ownerRef)
+}
+
+func TestFetchSensorDeploymentOwnerRef_ReplicaSetNotFound(t *testing.T) {
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), sensorPodName, sensorNamespace,
+		fake.NewSimpleClientset(createTestPod(sensorPodName, sensorNamespace, replicaSetName)), backoff)
+
+	assert.Error(t, err)
+	assert.Nil(t, ownerRef)
+}
+
+func TestFetchSensorDeploymentOwnerRef_UnexpectedPodOwners(t *testing.T) {
+	// create a pod with multiple owners
+	pod := createTestPod(sensorPodName, sensorNamespace, replicaSetName)
+	pod.OwnerReferences = append(pod.OwnerReferences, metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "Kind",
+		Name:       "second-owner",
+	})
+
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), sensorPodName, sensorNamespace,
+		fake.NewSimpleClientset(pod), backoff)
+
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "has unexpected owners")
+	assert.Nil(t, ownerRef)
+}
+
+func TestFetchSensorDeploymentOwnerRef_UnexpectedReplicaSetOwners(t *testing.T) {
+	// create a ReplicaSet with multiple owners
+	pod := createTestPod(sensorPodName, sensorNamespace, replicaSetName)
+	replicaSet := createTestReplicaSet(replicaSetName, sensorNamespace, deploymentName)
+	replicaSet.OwnerReferences = append(pod.OwnerReferences, metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "Kind",
+		Name:       "second-owner",
+	})
+
+	ownerRef, err := FetchSensorDeploymentOwnerRef(context.TODO(), sensorPodName, sensorNamespace,
+		fake.NewSimpleClientset(pod, replicaSet), backoff)
+
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "has unexpected owners")
```

Comment:
Ideally, we would rewrite the errors in a way, so that we can assert using `assert.ErrorIs`.
As a plan minimum, we should also assert on the words `pod` and `replicaset`, as the errors returned here and in `TestFetchSensorDeploymentOwnerRef_UnexpectedPodOwners` are different.

Reply:
Agreed about the errors, but I moved the from a file to another without modifying it, so I wanted to keep it that way. Could do this in a follow-up, or could do it here as well.